### ISSUE TITLE
Create documentation template

### ISSUE THREAD

**Issue by ChristinaWa** [dboehmer/coocook#17]
Think about how to document the project and which elements to provide additionally.

**Comment by dboehmer** [comment]
Documentation for end users or developers?

**Comment by ChristinaWa** [comment]
I meant for developers.

**Comment by dboehmer** [comment]
@glauschwuffel just mentioned that he likes to write documentation for Perl projects, maybe also for Coocook.

**Comment by glauschwuffel** [comment]
> @glauschwuffel just mentioned that he likes to write documentation for Perl projects, maybe also for Coocook.

Yes, I do.

So ... this ticket is about onboarding documentation for developers?

**Comment by dboehmer** [comment]
Yes. We will eventually also need user documentation but that has lower priority and UI is going to change. So let’s start with more developer documentation.

I thought of creating https://doc.coocook.org/ or similar and drop some useful stuff there:
* HTML output of Doxygen
* visualization of database design (how?)
* POD of our modules converted to HTML
* README.md as HTML
* additional HTML documents?

Questions we should probably cover:
* How to install and run Coocook? (README.md already contains most basic information)
* How to upgrade to newer versions?
* How to work with Docker? (@moseschmiedel is working on a Dockerfile for Coocook)

Everyone may feel free to add anything missing.

@glauschwuffel Is there anything in the list that you want to start with? Do you need information from us or should we have a hack session together to get you started?

**Comment by dboehmer** [comment]
Additional topic to cover:
* precedence logic of configuration files: `coocook_local.*` overrides `coocook.*` overrides `lib/Coocook.pm`
* configuration options in the config file (preferably `coocook_local.yaml`)&mdash;best documentation we currently have are the default values and comments in `lib/Coocook.pm`
